flink-libraries/flink-gelly-examples/src/main/java/org/apache/flink/graph/drivers/input/GridGraph.java

```diff
@@ -82,11 +81,27 @@ public void configure(ParameterTool parameterTool) throws ProgramParametrization
 
 	@Override
 	public String getIdentity() {
-		return getName() + " (" + dimensions + ")";
+		return getTypeName() + " " + getName() + " (" + dimensions + ")";
 	}
 
 	@Override
-	public Graph<LongValue, NullValue, NullValue> create(ExecutionEnvironment env) {
+	protected long vertexCount() {
+		// in Java 8 use Math.multiplyExact(long, long)
```

Comment:
Rather than the current implementation using `BigInteger`, Java 8's `Math.multiplyExact` detects and throws an exception on overflow. I was noting for the future when the Flink codebase targets Java 8.